### DISCUSSION TITLE
R做meta分析用forestplot包绘制森林图出现报错，求助

### DISCUSSION THREAD

**Discussion by JLR95** [ShixiangWang/self-study#85] [Q&A] [answered]
翔哥你好，我在用R做meta分析来绘制森林图，使用了forestplot包，出现了下面图片上的报错，后面把我的一整套文件发给我师妹去跑图，她们都可以出来，只有我自己的电脑会报错跑不出图，希望翔哥帮忙看下什么问题，万分感谢！

Error in assertAndRetrieveTidyValue(x, lower) : 
  You have not provided an argument and the data frame does not have a 'lower' column: V1, V2, V3, V4, V5, V6, V7, V8, V9

**Comment by ShixiangWang** [comment, marked as answer]
有没有检查过R包的版本呢？

**Reply by JLR95** [reply]
我后续对R包进行了更新，还是会报同样的错，我是不是需要卸载重新安装一下呢

**Reply by ShixiangWang** [reply]
> 我后续对R包进行了更新，还是会报同样的错，我是不是需要卸载重新安装一下呢

如果是更新成功了还有这样的问题的话，那有可能是依赖包的问题，或者说你用的R版本是否与你师妹使用的不一样呢？

目前的建议是使用 R 的 debug 函数层层调试报错的函数，看为什么这个 assertXXX 会报错。这个如果你没有经验的话，建议你学习一下，这个还是很有用的。

大致这样操作，假设你使用 `forestplot()` 报错了：

```R
# 先 debug 这个函数
debug(forestplot)

# 然后运行你报错的绘图函数
forestplot(xxx)

# 它会自动进入函数，你按步调试它
# 如果发现是其中一个函数报错，
# 那么继续利用 debug 进入其中步步查看
```

**Reply by JLR95** [reply]
非常感谢翔哥的回复！！我确实是个R语言小白，对R的使用不够了解。我按照您的操作进行了一下，进入了函数后出来下面这样的界面
function (...) 
{
  UseMethod("forestplot")
}
顶端还有提示Debug location is approximate because the source is not available.
接下来我该进一步怎么操作呢

**Reply by ShixiangWang** [reply]
一般一直敲回车就会步进了。就是一行一行运行内部的源代码。
